builtin/providers/azurerm/resource_arm_resource_group.go

```diff
@@ -0,0 +1,162 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/azure-sdk-for-go/arm/resources"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// resourceArmResourceGroup returns the *schema.Resource
+// associated to resource group resources on ARM.
+func resourceArmResourceGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmResourceGroupCreate,
+		Read:   resourceArmResourceGroupRead,
+		Exists: resourceArmResourceGroupExists,
+		Delete: resourceArmResourceGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArmResourceGroupName,
```

Comment:
Times may have changed and I might be mistaken now, but i distinctly remember @phinze asking me drop validation of this kind on Terraform's side on and old PR of mine because, as he argued, it should be the API's problem to validate its input as it may deem fit...